pkg/plugins/runtime/k8s/controllers/pod_controller.go

```
@@ -286,6 +286,10 @@ func (r *PodReconciler) createOrUpdateDataplane(
 	})
 	log := r.Log.WithValues("pod", kube_types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name})
 	if err != nil {
+		if errors.Is(err, context.Canceled) {
+			return nil
```

The thing I'm not sure about: does context.Canceled mean we need to retry?

---

I think we should bubble up the error